src/main/java/org/openstreetmap/atlas/proto/adapters/ProtoAtlasMetaDataAdapter.java

```diff
@@ -0,0 +1,108 @@
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Implements the {@link ProtoAdapter} interface to connect {@link AtlasMetaData} and
+ * {@link ProtoAtlasMetaData}.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasMetaDataAdapter implements ProtoAdapter
+{
+    private static final ProtoTagListConverter PROTOTAG_LIST_CONVERTER = new ProtoTagListConverter();
+
+    @Override
+    public ProtoSerializable deserialize(final byte[] byteArray)
+    {
+        ProtoAtlasMetaData protoAtlasMetaData = null;
+        try
+        {
+            protoAtlasMetaData = ProtoAtlasMetaData.parseFrom(byteArray);
+        }
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error encountered while parsing protobuf bytestream",
+                    exception);
+        }
+
+        final AtlasSize atlasSize = new AtlasSize(protoAtlasMetaData.getEdgeNumber(),
+                protoAtlasMetaData.getNodeNumber(), protoAtlasMetaData.getAreaNumber(),
+                protoAtlasMetaData.getLineNumber(), protoAtlasMetaData.getPointNumber(),
+                protoAtlasMetaData.getRelationNumber());
+
+        final boolean original = protoAtlasMetaData.getOriginal();
+        final String codeVersion = protoAtlasMetaData.getCodeVersion();
+        final String dataVersion = protoAtlasMetaData.getDataVersion();
+        final String country = protoAtlasMetaData.getCountry();
+        final String shardName = protoAtlasMetaData.getShardName();
+        final Map<String, String> tags = PROTOTAG_LIST_CONVERTER
+                .convert(protoAtlasMetaData.getTagsList());
+
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, original, codeVersion,
+                dataVersion, country, shardName, tags);
+
+        return atlasMetaData;
+    }
+
+    @Override
+    public byte[] serialize(final ProtoSerializable serializable)
+    {
+        if (!(serializable instanceof AtlasMetaData))
+        {
+            throw new CoreException(
+                    "Invalid ProtoSerializable type was provided to {}: cannot serialize {}",
+                    this.getClass().getName(), serializable.getClass().getName());
+        }
+        final AtlasMetaData atlasMetaData = (AtlasMetaData) serializable;
+
+        final ProtoAtlasMetaData.Builder builder = ProtoAtlasMetaData.newBuilder();
+        builder.setEdgeNumber(atlasMetaData.getSize().getEdgeNumber());
+        builder.setNodeNumber(atlasMetaData.getSize().getNodeNumber());
+        builder.setAreaNumber(atlasMetaData.getSize().getAreaNumber());
+        builder.setLineNumber(atlasMetaData.getSize().getLineNumber());
+        builder.setPointNumber(atlasMetaData.getSize().getPointNumber());
+        builder.setRelationNumber(atlasMetaData.getSize().getRelationNumber());
+        builder.setOriginal(atlasMetaData.isOriginal());
+
+        final Optional<String> codeVersionOptional = atlasMetaData.getCodeVersion();
```

Comment:
Instead of extracting in a local variable you can also use a lambda

```
atlasMetaData.getCodeVersion().ifPresent(value -> {
  builder.setCodeVersion(value);
});
```

Reply:
Good tip. I am new to using Optional. Thanks!